### PR TITLE
tools: Use removeBinding command in LSP

### PR DESCRIPTION
**Pull request by hunger**
Use the "removeBinding" command in the LSP to remove properties.

Note that this adds an "Add" button in addition to the remove button: This prevents the column form vanishing when no properties are set. This button is not wired up to anything yet though: Making that work is the next step:-)

### CODE REVIEW

**Review by ogoffart — COMMENTED** [review]
There is a bunch of console.log remaining

File: editors/vscode/src/common.ts [hunk 1]
```diff
@@ -86,11 +92,25 @@ export class PropertiesViewProvider implements vscode.WebviewViewProvider {
                         });
                     });
                     break;
+                case "remove_property":
+                    remove_property(
+                        data.document,
+                        data.element_range,
+                        data.property_name,
+                    )
+                        .then((_) => {
```
Comment (tronical) [inline editors/vscode/src/common.ts:101]: This and the catch seems weird. Why is this needed?

File: editors/vscode/src/common.ts [hunk 2]
```diff
@@ -254,114 +272,32 @@ export class PropertiesViewProvider implements vscode.WebviewViewProvider {
...
-                color: var(--error-bg);
-            }
-        `;
+        const styleUri = webview.asWebviewUri(
+            vscode.Uri.joinPath(this._extensionUri, "css", "content.css"),
+        );
+        const codiconsUri = webview.asWebviewUri(
+            vscode.Uri.joinPath(
+                this._extensionUri,
+                "node_modules",
```
Comment (tronical) [inline editors/vscode/src/common.ts:281]: Fingers crossed this also works in the extension build for the web. Might very well :)

File: tools/online_editor/src/properties_client.ts [hunk 3]
```diff
@@ -45,6 +45,23 @@ export async function change_property(
     return new Promise((accept) => accept({ diagnostics: [] }));
 }
 
+export async function remove_property(
```
Comment (tronical) [inline tools/online_editor/src/properties_client.ts:48]: Sometimes this is called remove_property in the code and sometimes removeBinding. I think it's clearly the latter, I suggest to rename the function etc.

File: tools/online_editor/src/shared/properties.ts [hunk 4]
```diff
@@ -284,11 +310,26 @@ export class PropertiesView {
                         );
                     }
                 });
+
+                delete_button.className = this.#remover_icon_class;
+                if (element_range != null) {
+                    delete_button.addEventListener("click", () => {
```
Comment (tronical) [inline tools/online_editor/src/shared/properties.ts:316]: I wonder if the closure should also take the event argument and call at least stopPropagation() on it.

**Review by tronical — COMMENTED** [review]
Looks overall good to me. What do you think about the suggestions?

File: editors/vscode/src/common.ts [hunk 5]
```diff
@@ -85,12 +91,23 @@ export class PropertiesViewProvider implements vscode.WebviewViewProvider {
...
                         });
                     });
+                    break;
+                case "remove_binding":
+                    remove_binding(
+                        data.document,
+                        data.element_range,
+                        data.property_name,
+                    ).catch((_) => {
+                        // catch this to avoid errors showing up in the console
```
Comment (tronical) [inline editors/vscode/src/common.ts:101]: Hm, if we're sending a command to our LSP, don't we want to know somehow when something goes wrong - for diagnostic purposes? Like via a console.log at least, not a popup?
